pgjdbc/src/test/java/org/postgresql/test/jdbc42/SetObject310Test.java

```diff
@@ -0,0 +1,307 @@
+package org.postgresql.test.jdbc42;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.sql.Types;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+
+import org.postgresql.test.TestUtil;
+
+import junit.framework.TestCase;
+
+public class SetObject310Test extends TestCase {
+
+    private Connection con;
+
+    public SetObject310Test(String name) {
+        super(name);
+    }
+
+    protected void setUp() throws Exception {
+        con = TestUtil.openDB();
+        TestUtil.createTable(con, "table1", "timestamp_without_time_zone_column timestamp without time zone,"
+                + "timestamp_with_time_zone_column timestamp with time zone,"
+                + "date_column date,"
+                + "time_without_time_zone_column time without time zone,"
+                + "time_with_time_zone_column time with time zone"
+                );
+    }
+
+    protected void tearDown() throws SQLException {
+        TestUtil.dropTable(con, "table1");
+        TestUtil.closeDB(con);
+    }
+    
+    /**
+     * Test the behavior setObject for timestamp columns.
+     */
+    public void testSetOffsetDateTimeWithType() throws SQLException {
+        PreparedStatement ps = con.prepareStatement(TestUtil.insertSQL("table1", "timestamp_with_time_zone_column", "?"));
+        OffsetDateTime data = OffsetDateTime.parse("2007-12-03T10:15:30+01:00");
+        try {
+            ps.setObject(1, data, Types.TIMESTAMP_WITH_TIMEZONE);
+            assertEquals(1, ps.executeUpdate());
+        } finally {
+            ps.close();
+        }
+        
+        Timestamp actual;
+        Statement st = con.createStatement();
+        try {
+            ResultSet rs = st.executeQuery(TestUtil.selectSQL("table1", "timestamp_with_time_zone_column"));
+            try {
+                assertNotNull(rs);
+                
+                assertTrue(rs.next());
+                actual = rs.getTimestamp(1);
+            } finally {
+                rs.close();
+            }
+        } finally {
+            st.close();
+        }
```

Comment:
@marschall , can you factor out "create statement -> execute -> getTimestamp" to a private method?
It seems you have the same pattern in quite a few of test methods.